### PR TITLE
Add liquidations to trades

### PR DESCRIPTION
**Pull request by Tburm**
Add liquidations as a new trade type. Liquidations are initially created during the `PositionModified` event in order to capture to trade direction (long/short), and is updated during the `PositionLiquidated` event with the trade size and fee.

### CODE REVIEW

File: src/futures.ts [hunk 1]
```diff
@@ -163,11 +144,29 @@ export function handlePositionModified(event: PositionModifiedEvent): void {
...
+  } else if (event.params.tradeSize.isZero() && event.params.size.isZero() && event.params.margin.isZero()) {
+    // it's a liquidation
+    let tradeEntity = new FuturesTrade(event.transaction.hash.toHex() + '-' + event.logIndex.toString());
+    tradeEntity.timestamp = event.block.timestamp;
+    tradeEntity.account = event.params.account;
+
+    // create a placeholder for trade size as long/short
+    // this will help figure out the trade direction during a liquidation
+    // since the position size will be set to zero before the PositionLiquidated event
+    tradeEntity.size = positionEntity.size.gt(ZERO) ? BigInt.fromI32(-1) : BigInt.fromI32(1);
```
Comment (brossetti1) [inline src/futures.ts:156]: nice solution here

File: src/futures.ts [hunk 2]
```diff
@@ -270,6 +262,14 @@ export function handlePositionLiquidated(event: PositionLiquidatedEvent): void {
     positionEntity.isLiquidated = true;
     positionEntity.save();
   }
+  if (tradeEntity) {
+    tradeEntity.size = event.params.size.times(tradeEntity.size);
```
Comment (brossetti1) [inline src/futures.ts:266]: maybe a comment here would help too

```suggestion
    // size will be either -1 or 1 in this scenario (see handlePositionModified for details)
    tradeEntity.size = event.params.size.times(tradeEntity.size);
```

**Review by brossetti1 — CHANGES_REQUESTED** [review]
just curious - i thought we talked about doing the entire tradeEntity in the `handlePositionLiquidated` event - do we not have enough info on `PositionLiquidatedEvent`?

nice job on this - will approve after response

**Comment by Tburm** [comment]
@brossetti1 That's correct, I was able to put the whole thing into the `PositionLiquidated` event but there is no way to tell which direction the trade goes. `PositionLiquidated` is called after `PositionModified`. `PositionModified` for a liquidation sets all values to zero (margin, size, trade size) so it's impossible to derive direction inside of `PositionLiquidated`.

**Comment by platschi** [comment]
https://github.com/Kwenta/kwenta/pull/833  has been approved and merged to frontend, so this PR is good to go.

**Review by brossetti1 — APPROVED** [review]

**Pull request merged**